src/mame/sega/segas32_v.cpp

```diff
@@ -744,8 +740,8 @@ void segas32_state::update_tilemap_zoom(screen_device &screen, segas32_state::la
 	srcy += (m_videoram[0x1ff14/2 + 4 * bgnum] & 0xfe00) << 4;
 
 	/* then account for the destination center coordinates */
-	srcx_start -= util::sext(m_videoram[0x1ff30/2 + 2 * bgnum], 10) * srcxstep;
-	srcy -= util::sext(m_videoram[0x1ff32/2 + 2 * bgnum], 9) * srcystep;
+	srcx_start -= util::sext(m_videoram[0x1ff30/2 + 2 * bgnum], (dstxstep != 0x200)?10:9) * srcxstep;
+	srcy -= util::sext(m_videoram[0x1ff32/2 + 2 * bgnum], (dstystep != 0x200)?10:9) * srcystep;
```

Comment:
If this is going to be merged, it needs a comment here to note that this is effectively a per-game hack masquerading as a global hack to fix this one case.  Also, please don’t code like your space bar is broken.

Reply:
I've updated the code with a more detailed note and added some spacing.

Reply:
Does the hw not have a zoom flag?
I see the same thing here, where it thinks zoom is enabled only if that value is not 0x200:

	/* enable this code below to display zoom information */
#if 0
	if (dstxstep != 0x200 || dstystep != 0x200)
	(...)

If a value of 0x200 really disables zoom and there is no specific flag for it, then no I don't think it's a hack.

Reply:
It’s not the way it’s checking for zoom being enabled that looks like a hack.  I still think ignoring one bit of the value unless zoom is enabled looks a lot like a hack, and I’d want to see some pretty convincing evidence to the contrary to believe otherwise.

Reply:
The code originally ran on 9 for all games before it was changed to 10 as a "temporary patch" for radr 20 something years ago. I wouldn't say we are ignoring a bit at 9...I would say it just uses more resolution when zooming.

It explains why all the games seem to suggest there's 9 bits of resolution and explains why the couple of outliers would use more. It tests working with all games at least...

I did update the PR to note that it might be a hack and explained to two most relevant cases it aims to correct. So hopefully that satisfies both sides to this.